Realm.Shared/RealmObject.cs

```diff
@@ -84,163 +84,162 @@ internal class Metadata
 
         internal void _CopyDataFromBackingFieldsToRow()
         {
-            Debug.Assert(this.IsManaged, "Can't copy from unmanaged object.");
+            Debug.Assert(IsManaged, "Can't copy from unmanaged object.");
             _metadata.Helper.CopyToRealm(this);
         }
 
         #region Getters
 
         protected string GetStringValue(string propertyName)
         {
-            Debug.Assert(_realm != null, "Object is not managed, but managed access was attempted");
+            Debug.Assert(IsManaged, "Object is not managed, but managed access was attempted");
 
-            return NativeTable.GetString(_metadata.Table, _metadata.ColumnIndices[propertyName], _rowHandle.RowIndex);
+            return NativeTable.GetString(_objectHandle, _metadata.PropertyIndices[propertyName]);
```

Comment:
I think the `Get`/`Set` family of methods should now be moved to `ObjectHandle`.